ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java

```diff
@@ -180,7 +181,21 @@ private void appendIndex(QueryEntity queryEntity, AssociationKeyMetadata associa
 			fields.put( realColumnName, true );
 		}
 		queryIndex.setFields( fields );
-		queryIndex.setName( queryEntity.getTableName() + '_' + org.hibernate.ogm.util.impl.StringHelper.join( fields.keySet(), "_" ) );
+		String indexName = queryEntity.getTableName() + '_' + UUID.nameUUIDFromBytes( org.hibernate.ogm.util.impl.StringHelper.join( fields.keySet(), "_" ).getBytes() );
+		Class<?> tableClass = context.getTableEntityTypeMapping().get( associationKeyMetadata.getAssociatedEntityKeyMetadata().getEntityKeyMetadata().getTable() );
+		javax.persistence.Table tableAnnotation = tableClass.getAnnotation( javax.persistence.Table.class );
+		if ( tableAnnotation != null && tableAnnotation.indexes().length > 0 ) {
+			for ( javax.persistence.Index index: tableAnnotation.indexes() ) {
+				if ( !index.name().isEmpty() ) {
+					indexName = index.name();
+					break;
+				}
+			}
+		}
+		if ( indexName.getBytes().length > 255 ) {
+			throw log.indexNameTooLong( indexName, queryEntity.getTableName() );
```

Comment:
Would it be possible to collect all the indexes with an invalid name and then throw an exception at the end?
It's not a critical problem but when there is more than one error, the user can see all the problems in one go instead of 
 fix and rerun the application everytime an index fails the validation.